### PR TITLE
Resolve the dependencies for moving BlockStore to a separate project

### PR DESCRIPTION
**Pull request by quantumagi**
- Remove LoggingConfiguration's dependency on BlockStore
- Remove NodeSettings dependency on BlockStore

### CODE REVIEW

File: Stratis.Bitcoin/Features/BlockStore/BlockStoreFeature.cs [hunk 1]
```diff
@@ -121,6 +124,8 @@ public static partial class IFullNodeBuilderExtensions
     {
         public static IFullNodeBuilder UseBlockStore(this IFullNodeBuilder fullNodeBuilder)
         {
+            LoggingConfiguration.RegisterFeatureNamespace<BlockStoreFeature>("db");
```
Comment (dangershony) [inline Stratis.Bitcoin/Features/BlockStore/BlockStoreFeature.cs]: Hmm I think db refers to to CoinView this is only BlockStore

**Comment by dangershony** [comment]
Can you please resolve conflicts?

**Comment by bokobza** [comment]
I'll take care of this one @dangershony

File: Stratis.Bitcoin/Features/BlockStore/BlockStoreFeature.cs (hunk 1 above)
Comment (quantumagi) [reply Stratis.Bitcoin/Features/BlockStore/BlockStoreFeature.cs]: Compared to the code being replaced this should preserve the current behaviour - i.e. it does seem like "db" has been used in conjunction with the BlockStore feature up to now.

File: Stratis.Bitcoin/Configuration/Settings/StoreSettings.cs [hunk 2]
```diff
@@ -14,12 +17,32 @@ public class StoreSettings
...
+        public StoreSettings(Action<StoreSettings> callback)
+            :this()
+        {
+            this.callback = callback;
+        }
+
+        public StoreSettings(NodeSettings nodeSettings, Action<StoreSettings> callback = null)
+            :this(callback)
+        {
+            Load(nodeSettings);
```
Comment (bokobza) [inline Stratis.Bitcoin/Configuration/Settings/StoreSettings.cs]: Please use "this." to mark instance methods.

Comment (quantumagi) [reply Stratis.Bitcoin/Configuration/Settings/StoreSettings.cs]: Done

**Comment by bokobza** [comment]
Great stuff, thank you.
I saw that the test ReorganiseBlockRepository_WithBlockRepositoryAndChainOutofSync_ReorganiseBlocks_InMemory is failing. Is that something that you can fix?

**Comment by quantumagi** [comment]
@bokobza, it is one of those tests that succeeds when run on its own and would be failing intermittently. Perhaps this can be fixed in a separate PR along with other similar issues?

**Comment by bokobza** [comment]
fine, thanks mate.

**Pull request merged**